config.go

```diff
@@ -1,9 +1,35 @@
 package ynabber
 
 import (
+	"encoding/json"
 	"time"
 )
 
+const DateFormat = "2006-01-02"
+
+type Date time.Time
+
+// Decode implements `envconfig.Decoder` for Date to parse string to time.Time
+func (date *Date) Decode(value string) error {
+	time, err := time.Parse(DateFormat, value)
+	if err != nil {
+		return err
+	}
+	*date = Date(time)
+	return nil
+}
+
+type AccountMap map[string]string
+
+// Decode implements `envconfig.Decoder` for AccountMap to decode JSON properly
+func (accountMap *AccountMap) Decode(value string) error {
+	err := json.Unmarshal([]byte(value), &accountMap)
+	if err != nil {
+		return err
```

Comment:
Same here about the sensible error message.

Reply:
It fails somewhat nice
```
$ go run cmd/ynabber/main.go
2022/11/17 17:17:34 envconfig.Process: assigning NORDIGEN_NORDIGEN_ACCOUNTMAP to AccountMap: converting 'fooo' to type ynabber.AccountMap. details: invalid character 'o' in literal false (expecting 'a')
exit status 1
```